src/Raven.Server/Rachis/Candidate.cs

```diff
@@ -61,7 +66,7 @@ private void Run()
 
                     if (clusterTopology.Members.Count == 1)
                     {
-                        CastVoteForSelf(ElectionTerm + 1, "Single member cluster, natural leader");
+                        await CastVoteForSelfAsync(ElectionTerm + 1, "Single member cluster, natural leader");
```

Comment:
Not happy with this. 
It violates an important consideration for us, that we can always see the operations of the leader/ candidate / follower in their thread.
This way, we may return to _another_ thread here, and the candidate thread is just stalled.